src/lib.rs

```diff
@@ -38,12 +40,26 @@
 #![deny(unused_must_use)]
 #![deny(broken_intra_doc_links)]
 
-extern crate core;
+#[cfg(not(any(feature = "std", feature = "no-std")))]
+compile_error!("at least one of the `std` or `no-std` features must be enabled");
+
+#[cfg(feature = "no-std")]
+#[macro_use]
+pub extern crate alloc;
+#[cfg(feature = "no-std")]
+pub extern crate core2;
+
+#[cfg(any(feature = "std", test))]
+extern crate core; // for Rust 1.29 and no-std tests
 
 // Re-exported dependencies.
 #[macro_use] pub extern crate bitcoin_hashes as hashes;
 pub extern crate secp256k1;
 pub extern crate bech32;
+
+#[cfg(feature = "no-std")]
+pub extern crate hashbrown;
```

Comment:
Is there a reason to re-export this (and alloc/core2, above)?

Reply:
no, will fix